src/main/java/com/powsybl/study/server/StudyService.java

```diff
@@ -304,4 +289,39 @@ void setGeoDataServerRest(RestTemplate geoDataServerRest) {
         this.geoDataServerRest = Objects.requireNonNull(geoDataServerRest);
     }
 
+    void setNetworkMapServerRest(RestTemplate networkMapServerRest) {
+        this.networkMapServerRest = Objects.requireNonNull(networkMapServerRest);
+    }
+
+    String getSubstationsMapData(UUID networkUuid) {
+        Map<String, Object> urlParams = new HashMap<>();
+        urlParams.put(NETWORK_UUID, networkUuid);
+
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(networkMapServerBaseUri + "/" + NETWORK_MAP_API_VERSION +
```

Comment:
You can use UriComponentsBuilder.fromPath to take advantage of already configured base uri (see constructor). Also you don't need for simple template to build an intermediate map for variables.
=> See CaseDataSourceClient code

Reply:
Done